### ISSUE TITLE
FIO: Add support for Payment Request

### ISSUE THREAD

**Issue by catenocrypt** [trustwallet/wallet-core#850]
As part of FIO support https://github.com/trustwallet/internal/issues/213 , add handling of Payment Requeststo FIO Signer.
Payment Request uses the following two signed APIs:
- new_funds_request
- reject_funds_request

Info: the content field is an encrypted payload, requires additional handling, see:  https://github.com/fioprotocol/fiojs/blob/master/docs/message_encryption.md
Info: There are additional related API calls, but those are not signed:  get_sent_fio_requests, get_pending_fio_requests.

**Comment by catenocrypt** [comment]
Note: PR #848 .

**Comment by catenocrypt** [comment]
Implementation done for NewFundsRequest -- shared secret derivation, ecnryption (AES CBC with HMAC), serialization, message fields -- in PR #848 (branch ar/fio).

Remains to validate in the testnet (create new funds request, query it back).  First test was not successful, to check content encoding (currently is hex string), double check implementation.  I was not yet able to do a NewFundsRequest with existing code (fiojs has no support, to check fiosdk_typescript).

**Comment by catenocrypt** [comment]
Update: I was still not able to create a valid new_payment_request.  Developer doc shows now new chainCode parameters, but even adding that did not help.  Also, there is contradictory info if tpId is needed or not, see screenshot below.

The error I get:
```json
{ \n  \"type\": \"invalid_transaction\",\n  \"message\": \"Signed transaction is not valid or is not formatted properly\"\n}
```

![image](https://user-images.githubusercontent.com/13562139/75794460-21eba380-5d71-11ea-9e4f-bc956dff7a7b.png)

**Comment by catenocrypt** [comment]
Still no success, generated NewFundsRequest tx is not accepted by v0.9 testnet.
I code reviewed the fioprotocol/fio code, and code in WC, tried a few variation, with no success.

I will close this for now, as a known issue; funds request is not high prio, it should be fine to solve after mainnet launch.

**Comment by catenocrypt** [comment]
With the help FIO developers, I was able to figure out the issue with NewFundsRequest.  
Differences were:
- base64 encoding of encrypted content (I used plain hex string coding).
- actor serialization was EOS::Name serialization, as in other places, but I see correct is as string

I've manages to create valid NewFundsRequest, and also query it back.
Solved in the PR.

**Comment by catenocrypt** [comment]
I did some free test again, created register_fio_address, add_pub_address, and new_funds_request transactions, all accepted by the testnet.  Funds request can be queried back (currently there are 2, see below).

Issue can be closed when PR #883 is merged.

```
curl -v -d '{"fio_public_key": "FIO7uMZoeei5HtXAD24C4yCkpWWbf24bjYtrRNjWdmGCXHZccwuiE", "limit": 100, "offset": 0}' http://testnet.fioprotocol.io/v1/chain/get_pending_fio_requests


{"requests":[
  {
    "fio_request_id":531562,
    "payer_fio_address":"adam@fiotestnet",
    "payee_fio_address":"adam@fiotestnet",
    "payer_fio_public_key":"FIO7uMZoeei5HtXAD24C4yCkpWWbf24bjYtrRNjWdmGCXHZccwuiE",
    "payee_fio_public_key":"FIO7uMZoeei5HtXAD24C4yCkpWWbf24bjYtrRNjWdmGCXHZccwuiE",
    "content":"AAECAwQFBgcICQoLDA0OD60C6IxdN4Q6qhhscj1CKt8cDVXehC0c70YKhi3Z3IS9o9ggznzKcGLxAdTQAF05W6DsBG0omEzGTN3BQj4tS5Lc9xu6BWhZ4pRbE8TjikDmFZIjIf+PWc2ZnJb+wOh5rm86/mxbeCoEZ+Zx1Id1ndoDYDQadbEeSnD4UytEDsO0",
    "time_stamp":"2020-03-09T15:43:24"
  },
  {
    "fio_request_id":531650,
    "payer_fio_address":"adam2@fiotestnet",
    "payee_fio_address":"adam@fiotestnet",
    "payer_fio_public_key":"FIO7uMZoeei5HtXAD24C4yCkpWWbf24bjYtrRNjWdmGCXHZccwuiE",
    "payee_fio_public_key":"FIO7uMZoeei5HtXAD24C4yCkpWWbf24bjYtrRNjWdmGCXHZccwuiE",
    "content":"AAECAwQFBgcICQoLDA0OD60C6IxdN4Q6qhhscj1CKt8cDVXehC0c70YKhi3Z3IS9TvgnQnDlOtvUXwDgD5LQ9yf7Eq7clrs0kf1/6AxhOo9O0C3GCbWAJKJLJ0PlwTXWz5do5T3hrgYYv3+FU+ju8rdsbedYS1UU2xR4DlU90FWUeOvDeg6aoFWbG/3za0/a",
    "time_stamp":"2020-03-10T08:42:27"
  }
],
"block_num":82330720,
"more":0
}
```

**Comment by catenocrypt** [comment]
Closed via #883 .
